addons/event_sale/views/event_views.xml

```diff
@@ -1,5 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
+        <record id="sale_order_action_view_from_event" model="ir.actions.act_window">
+            <field name="name">Sales Order</field>
+            <field name="res_model">sale.order</field>
+            <field name="view_id" ref="sale.view_order_tree"/>
+            <field name="domain">[('state', '!=', 'cancel'), ('order_line.event_id', '=', active_id)]</field>
+            <field name="context">{'create': 0}</field>
+        </record>
```

Comment:
Can't you replace that by a python method that fetches an existing action instead of creating a new one?